### PR TITLE
Azure workload cluster release 14.0.0 with Kubernetes 1.19

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

**Comment by MarcelMue** [comment]
This is the PR for 1.19 Azure. CNCF already passed, currently waiting for https://github.com/giantswarm/azure-operator/pull/1259 to proceed here.

File: azure/v14.0.0/release.diff [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Generated with:                                                  # Generated with:
+# devctl release create --name 13.0.2 --provider azure --base 1 |  # devctl release create --name 14.0.0 --base v13.0.2 --releases
```
Comment (tuommaki) [inline azure/v14.0.0/release.diff:2]: Should the base release now be `13.1.0` since that one is released?

File: azure/v14.0.0/release.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      reference: 0.1.0-2
+      releaseOperatorDeploy: true
+      version: 0.1.0
+    - name: cluster-operator
+      releaseOperatorDeploy: true
+      version: 0.23.20
+    - name: kubernetes
+      version: 1.19.4
+    - name: containerlinux
+      version: 2605.9.0
```
Comment (tuommaki) [inline azure/v14.0.0/release.yaml:49]: Latest stable flatcar is `2605.11.0`.

File: azure/v14.0.0/release.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - name: coredns
+      version: 1.6.5
+    - name: calico
```
Comment (whites11) [inline azure/v14.0.0/release.yaml:52]: What about etcd and calico? should be bump them as well?

current latest releases:
- etcd 3.4.14
- calico 3.16.6

Comment (MarcelMue) [reply azure/v14.0.0/release.yaml:54]: Will need to do some extra legwork for calico first:
```
{"caller":"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource/crud_resource.go:89","controller":"azure-operator-azureconfig-controller","event":"update","function":"GetDesiredState","level":"error","loop":"11","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/azureconfigs/2t7ut","resource":"blobobject","stack":{"annotation":"template.CloudConfigConfig.validation error: component `Calico` requires version following constraint `>= 3.10.0 < 3.16.0`, got `v3.16.6`","kind":"invalidConfigError","stack":[{"file":"/go/pkg/mod/github.com/giantswarm/k8scloudconfig/v10@v10.0.0/pkg/template/cloudconfig.go","line":35},{"file":"/root/project/service/controller/cloudconfig/cloud_config.go","line":159},{"file":"/root/project/service/controller/resource/blobobject/desired.go","line":161},{"file":"/go/pkg/mod/github.com/giantswarm/operatorkit/v4@v4.2.0/pkg/resource/wrapper/retryresource/crud_resource.go","line":82}]},"time":"2021-01-18T14:29:22.785689+00:00","version":"253373916"}
```

File: azure/v14.0.0/release.yaml [hunk 4]
```diff
@@ -7,54 +7,56 @@ metadata:
...
+      version: 1.19.4
+    - name: containerlinux
+      version: 2605.10.0
+    - name: coredns
+      version: 1.6.5
+    - name: calico
+      version: 3.15.3
+    - name: etcd
+      version: 3.4.14
+    - name: chart-operator
```
Comment (whites11) [inline azure/v14.0.0/release.yaml:58]: Why is this here? shouldn't it be in the apps section only?

Comment (MarcelMue) [reply azure/v14.0.0/release.yaml:58]: My mistake most likely, going to fix

File: azure/v14.0.0/release.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      reference: 5.2.0-9b3aea291daacdc6f61ad67c9a91c1163be347d1
+    - name: cert-operator
+      reference: 0.1.0-2
+      releaseOperatorDeploy: true
+      version: 0.1.0
+    - name: cluster-operator
+      releaseOperatorDeploy: true
+      version: 0.23.20
+    - name: kubernetes
+      version: 1.19.4
```
Comment (tuommaki) [inline azure/v14.0.0/release.yaml:51]: Is there a specific reason to go with `1.19.4` when latest one is `1.19.7` ?

Comment (MarcelMue) [reply azure/v14.0.0/release.yaml:51]: IIRC when we started only 1.19.4 was out - 1.19.7 should be fine now

Comment (tuommaki) [reply azure/v14.0.0/release.yaml:51]: Yeah, there were bunch of bugfixes for Azure disk related stuff (again) and few fixes for OOM / CPU burn situations.

Comment (whites11) [reply azure/v14.0.0/release.yaml:51]: wlll bump it

Comment (whites11) [reply azure/v14.0.0/release.yaml]: done

File: azure/v14.0.0/release.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    - name: azure-scheduled-events
+      version: 0.1.1
+  components:
+    - name: app-operator
+      releaseOperatorDeploy: true
+      version: 3.0.0
+    - name: azure-operator
+      catalog: control-plane-test-catalog
+      releaseOperatorDeploy: true
+      version: 5.2.2-dev
```
Comment (MarcelMue) [inline azure/v14.0.0/release.yaml:41]: Needs to be tagged

File: azure/v14.0.0/release.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+metadata:
+  creationTimestamp: null
+  name: v14.0.0
+spec:
+  apps:
+    - name: cert-exporter
+      version: 1.6.0
+    - name: chart-operator
+      version: 2.6.0
+    - componentVersion: 1.6.5
```
Comment (MarcelMue) [inline azure/v14.0.0/release.yaml:14]: CoreDNS is here and in components, should only appear once. I think this was already and issue in earlier releases.

File: azure/v14.0.0/release.yaml (hunk 6 above)
Comment (whites11) [reply azure/v14.0.0/release.yaml:41]: it will be tagged once tests are green

File: azure/v14.0.0/release.yaml (hunk 7 above)
Comment (whites11) [reply azure/v14.0.0/release.yaml:14]: removed

**Comment by fiunchinho** [comment]
With latest change in cluster-operator version, tests can't create a kubeconfig

```
Not yet available
It is not yet possible to create a key pair for this cluster, as the PKI backend is not yet available.
Please try again in a moment.
```

**Comment by fiunchinho** [comment]
There was a bug in cluster-operator `0.23.21`, so I just pushed a commit updating it to `0.23.22`.

**Comment by fiunchinho** [comment]
I want to trigger all tests at once and see what happens (some people just want to see the world burn)

File: azure/v14.0.0/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+# :zap: Giant Swarm Release v14.0.0 for Azure :zap:
+
+This is the first tenant cluster release to support Kubernetes 1.19 on Azure.
```
Comment (MartaD) [inline azure/v14.0.0/README.md:3]:
```suggestion
This is the first workload cluster release to support Kubernetes 1.19 on Azure.
```

File: azure/v14.0.0/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+Thanks for the [added support by `Azure`](https://azure.microsoft.com/en-us/updates/germany-west-central-availability-zones-now-generally-available/),
```
Comment (MartaD) [inline azure/v14.0.0/README.md:5]:
```suggestion
Thanks to the [added support by `Azure`](https://azure.microsoft.com/en-us/updates/germany-west-central-availability-zones-now-generally-available/),
```

File: azure/v14.0.0/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+with this release `Availability Zones` are supported for all `Workload Clusters` running in the `Germany West Central`
```
Comment (MartaD) [inline azure/v14.0.0/README.md:6]:
```suggestion
with this release `Availability Zones` are supported for all workload clusters running in the `Germany West Central`
```

File: azure/v14.0.0/README.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+region as well.
+
+Starting from this release, Azure `Workload Clusters` include by default a new application named `azure-scheduled-events`
```
Comment (MartaD) [inline azure/v14.0.0/README.md:9]:
```suggestion
Starting from this release, Azure workload clusters include by default a new application named `azure-scheduled-events`
```

File: azure/v14.0.0/README.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+that leverages the [Azure scheduled events](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/scheduled-events)
+feature to automatically drain a `Kubernetes` node when the underlying virtual instance is about to be terminated.
+This makes the cluster behaviour better when the `cluster autoscaler` scales down a node pool.
```
Comment (MartaD) [inline azure/v14.0.0/README.md:12]: how does it make it better? draining is faster?

**Review by MartaD — APPROVED** [review]

File: azure/v14.0.0/README.md (hunk 12 above)
Comment (whites11) [reply azure/v14.0.0/README.md:12]: much faster

File: azure/v14.0.0/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (whites11) [reply azure/v14.0.0/README.md]: Changed, thanks

File: azure/v14.0.0/README.md (hunk 12 above)
Comment (MartaD) [reply azure/v14.0.0/README.md:12]: This ensures the workload running on the node is handled gracefully when the `cluster autoscaler` scales down a node pool.

Maybe this?

Comment (whites11) [reply azure/v14.0.0/README.md]: much better, updated, thx

**Review by T-Kukawka — APPROVED** [review]
